reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java

```diff
@@ -1893,6 +1893,68 @@ public void channelRead(@NotNull ChannelHandlerContext ctx, @NotNull Object msg)
 				.isEqualTo("delay500delay1000");
 	}
 
+	@Test
+	public void testGracefulShutdownIssue3509() throws Exception {
+		CountDownLatch latch1 = new CountDownLatch(2);
+		CountDownLatch latch2 = new CountDownLatch(1);
+		CountDownLatch latch3 = new CountDownLatch(1);
```

Comment:
Consider naming the latches according to their function:
`latch1` -> `connectedLatch`
`latch2` -> `disposeSubscribedLatch`
`latch3` -> `responseReceivedLatch`

Reply:
fixed